Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -110,32 +118,54 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         boundStreamBufferSize: Int? = nil,
         logger: LogAgent
     ) {
+        self.bridgeBufferSize = bridgeBufferSize
+        self.boundStreamBufferSize = boundStreamBufferSize ?? bridgeBufferSize
+        self.buffer = Data(capacity: bridgeBufferSize)
+        self.readableStream = readableStream
+        self.logger = logger
+        (inputStream, outputStream) = Self.makeStreams(boundStreamBufferSize: self.boundStreamBufferSize, queue: queue)
+    }
+
+    func replaceStreams(completion: @escaping (InputStream?) -> Void) async {
```

Comment:
This method will be called by the HTTP client when a new body stream is needed.

The output stream is closed, both streams are replaced with new ones & configured, and the output stream is reopened.

Note that I never do anything to the input stream, only the URLSessionDataTask ever manipulates that stream.